components/camel-sql/src/main/java/org/apache/camel/component/sql/stored/template/ast/ParseHelper.java

```diff
@@ -21,29 +21,75 @@
 
 import org.apache.camel.component.sql.stored.template.generated.SSPTParserConstants;
 import org.apache.camel.component.sql.stored.template.generated.Token;
+import org.springframework.util.ClassUtils;
 import org.springframework.util.ReflectionUtils;
 
 public final class ParseHelper {
 
     private ParseHelper() {
     }
 
-    public static int parseSqlType(Token sqlType) {
+    public static int parseSqlType(Token sqlTypeToken) {
+
+        String sqlType = sqlTypeToken.toString();
 
         //if number then use it(probably Vendor spesific SQL-type)
-        if (sqlType.kind == SSPTParserConstants.NUMBER) {
-            return Integer.valueOf(sqlType.toString());
+        if (sqlTypeToken.kind == SSPTParserConstants.NUMBER) {
+            return Integer.valueOf(sqlType);
+        }
+
+        //if contains .
+        if (sqlType.contains(".")) {
+            String className;
+            String fieldName;
+            try {
+                className = sqlType.substring(0, sqlType.lastIndexOf("."));
+                fieldName = sqlType.substring(sqlType.lastIndexOf(".") + 1);
+            } catch (Exception ex) {
+                throw new ParseRuntimeException("Failed to parse class.field:" + sqlType);
+            }
+            try {
+                Class clazz = ClassUtils.forName(className, ParseHelper.class.getClassLoader());
+                return getFieldInt(clazz, fieldName);
+            } catch (ClassNotFoundException e) {
+                throw new ParseRuntimeException("Class for " + className + " not found", e);
+
+            }
         }
 
         //Loop-up from "Standard" types
-        Field field = ReflectionUtils.findField(Types.class, sqlType.toString());
+        return getFieldInt(Types.class, sqlType);
+
+
+    }
+
+    public static Integer parseScale(Token token) {
+        try {
+            String str = token.toString();
+            return Integer.valueOf(str.substring(1, str.length() - 1));
+        } catch (Exception ex) {
+            throw new ParseRuntimeException("Failed to parse scale from token:" + token.toString(), ex);
+        }
+    }
+
+    private static int getFieldInt(Class clazz, String sqlType) {
+        Field field = ReflectionUtils.findField(clazz, sqlType);
         if (field == null) {
-            throw new ParseRuntimeException("Field " + sqlType + " not found from java.procedureName.Types");
+            throw new ParseRuntimeException("Field " + sqlType + " not found from " + clazz.getName());
         }
         try {
             return field.getInt(Types.class);
         } catch (IllegalAccessException e) {
             throw new ParseRuntimeException(e);
         }
     }
+
+
+    public static String removeQuotes(String token) {
```

Comment:
There is a `removeLeadingAndEndingQuotes` method in camel-core from StringHelper which should be used instead